yt_dlp/extractor/servus.py

```diff
@@ -15,130 +12,97 @@ class ServusIE(InfoExtractor):
-                    'height': int_or_none(resource.get('height')),
-                })
+            'https://api-player.redbull.com/stv/servus-tv?timeZone=Europe/Berlin&videoId=%s' % video_id,
+            video_id, 'Downloading video JSON')
+        if 'videoUrl' not in video:
+            self._report_errors(video)
+            return None
+        formats, subtitles = self._extract_m3u8_formats_and_subtitles(
+            video['videoUrl'], video_id, 'mp4', entry_protocol='m3u8_native',
+            m3u8_id='hls')
 
-        attrs = {}
-        for attribute in video['attributes']:
-            if not isinstance(attribute, dict):
-                continue
-            key = attribute.get('fieldKey')
-            value = attribute.get('fieldValue')
-            if not key or not value:
-                continue
-            attrs[key] = value
-
-        title = attrs.get('title_stv') or video_id
-        alt_title = attrs.get('title')
-        description = attrs.get('long_description') or attrs.get('short_description')
-        series = attrs.get('label')
-        season = attrs.get('season')
-        episode = attrs.get('chapter')
-        duration = float_or_none(attrs.get('duration'), scale=1000)
+        season = video.get('season')
         season_number = int_or_none(self._search_regex(
             r'Season (\d+)', season or '', 'season number', default=None))
+        episode = video.get('chapter')
         episode_number = int_or_none(self._search_regex(
             r'Episode (\d+)', episode or '', 'episode number', default=None))
 
         return {
             'id': video_id,
-            'title': title,
-            'alt_title': alt_title,
-            'description': description,
-            'thumbnail': thumbnail,
-            'duration': duration,
-            'timestamp': unified_timestamp(video.get('lastPublished')),
-            'series': series,
-            'season': season,
+            'title': video.get('title'),
+            'description': self._get_description(video, video_id),
+            'thumbnail': video.get('poster'),
+            'duration': float_or_none(video.get('duration')),
+            'timestamp': unified_timestamp(video.get('currentSunrise')),
+            'series': season,
+            'season': video.get('season'),
             'season_number': season_number,
             'episode': episode,
             'episode_number': episode_number,
             'formats': formats,
+            'subtitles': subtitles,
         }
+
+    def _get_description(self, video, video_id):
+        info = self._download_json("https://backend.servustv.com/wp-json/rbmh/v2/media_asset/aa_id/%s?fieldset=page" % video_id,
+                                   video_id, fatal=False)
+        if not info:
+            return video.get('description')
+        description = info.get('stv_long_description') \
+            or info.get("stv_short_description") \
+            or video.get('description')
+        return description
```

Comment:
> [pukkandan](https://github.com/pukkandan) [on Jul 10, 2022](https://github.com/yt-dlp/yt-dlp/pull/4240#discussion_r917394552)
Actually, passing video here only to do a .get on it seems unnecessary. Do this instead:
```suggestion
    def _get_description(self, video_id):
        info = self._download_json(
            f'https://backend.servustv.com/wp-json/rbmh/v2/media_asset/aa_id/{video_id}?fieldset=page',
            video_id, fatal=False)
        return traverse_obj(info, 'stv_long_description', 'stv_short_description')
```
> and above

            'description': self._get_description(video_id) or video.get('description'),

Reply:
(Since this was multiple changes, had to amend this in a3883932)